### PR TITLE
add argparse2tool

### PR DESCRIPTION
**Pull request by bernt-matthias**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/argparse2tool/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+build:
+  number: 0
+  entry_points:
+    - argparse2tool_check_path = argparse2tool.check_path:main
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+  noarch: python
+
+requirements:
+  host:
+    - future
```
Comment (bgruening) [inline recipes/argparse2tool/meta.yaml:21]: only pip and python need to stay here

File: recipes/argparse2tool/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - argparse2tool
+    - argparse2tool.cmdline2cwl
+    - argparse2tool.cmdline2gxml
+    - click
+  commands:
+    - argparse2tool_check_path --help
+
+about:
+  home: "https://github.com/erasche/argparse2tool"
+  license: Apache Software
```
Comment (cbrueffer) [inline recipes/argparse2tool/meta.yaml:44]: This should be more specific, e.g. Apache-2.0

**Comment by bgruening** [comment]
galaxyxml is missing as package as it seems.

**Comment by bernt-matthias** [comment]
on the way https://github.com/bioconda/bioconda-recipes/pull/23348 :)

**Comment by bernt-matthias** [comment]
@BiocondaBot please fetch artifacts

**Comment by bernt-matthias** [comment]
@BiocondaBot please fetch artifacts

File: recipes/argparse2tool/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - argparse2tool_check_path = argparse2tool.check_path:main
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+  noarch: python
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - future
```
Comment (mbargull) [inline recipes/argparse2tool/meta.yaml:24]: Seeing https://github.com/hexylena/argparse2tool/commit/cdc93e5c1f164645045ecef40625e03b4806960d :
```suggestion
```
Plus, add `>=3.what-ever-is-appropriate` to the `python` dep in `run` as well as `host`.

Comment (bernt-matthias) [reply recipes/argparse2tool/meta.yaml:24]: Good idea.

**Comment by bernt-matthias** [comment]
Seems that we need one more iteration. Currently the module loading mechanism tries to load the original click/argparse module by checking if argparse2tool appears in the module path. If the conda package is installed into an environment with the name argparse2tool this does not work


https://github.com/hexylena/argparse2tool/pull/69

**Comment by bernt-matthias** [comment]
@BiocondaBot please fetch artifacts

**Comment by bernt-matthias** [comment]
@BiocondaBot please add label

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**